cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -335,19 +327,13 @@ class ConsumingSinkNode : public ExecNode, public BackpressureControl {
   void Resume() override { inputs_[0]->ResumeProducing(this, ++backpressure_counter_); }
 
   void StopProducing() override {
-    EVENT(span_, "StopProducing");
     if (input_counter_.Cancel()) {
       Finish(Status::OK());
     }
   }
 
   void InputReceived(ExecNode* input, ExecBatch batch) override {
-    EVENT(span_, "InputReceived", {{"batch.length", batch.length}});
-    util::tracing::Span span;
-    START_COMPUTE_SPAN_WITH_PARENT(
-        span, span_, "InputReceived",
-        {{"node.label", label()}, {"batch.length", batch.length}});
-
+    auto scope = TraceInputReceived(batch);
```

Comment:
If the sink applies backpressure (e.g. the datasetwriter), it does not result in an event being created on the span as is the case for the normal SinkNode here https://github.com/apache/arrow/pull/33738/files#diff-967cff6ef1964402635ac0769dece741ca0ed58bcefb8d669ecfe3ed8371998eR175
Is there a way to do this? For example, we could compare the value of `backpressure_counter_` before and after calling `Consume()`, but then we wouldn't be able to know if backpressure was applied or released.

Reply:
The backpressure is applied in the dataset writer (it has it's own queue).  However, we can add tracing events in the dataset writer.  I'm kind of interested now to see what a dataset writer trace looks like.  I will add something.

Reply:
Yes, I think that makes a lot of sense especially because the dataset writer submits tasks to the IO executor that can run in parallel. All that would otherwise be lost behind a single ConsumingSinkNode span